src/java/org/apache/cassandra/streaming/StreamSession.java

```diff
@@ -981,7 +987,7 @@ public void streamSent(OutgoingStreamMessage message)
         StreamTransferTask task = transfers.get(message.header.tableId);
         if (task != null)
         {
-            task.scheduleTimeout(message.header.sequenceNumber, 12, TimeUnit.HOURS);
+            task.scheduleTimeout(message.header.sequenceNumber, DatabaseDescriptor.getStreamTransferTaskTimeout(), TimeUnit.MILLISECONDS);
```

Comment:
do conversation at this layer, do not delegate that to DD

Reply:
conversation?

Reply:
you fixed this with `.toMilliseconds()`